Library/Homebrew/dev-cmd/boneyard-formula-pr.rb

```diff
@@ -0,0 +1,104 @@
+# Creates a pull request to boneyard a formula.
+#
+# Usage: brew boneyard-formula-pr [options...] <formula-name>
+#
+# Options:
+#   --dry-run:  Print what would be done rather than doing it.
+
+require "formula"
+require "utils/json"
+require "fileutils"
+
+module Homebrew
+  def boneyard_formula_pr
+    formula = ARGV.formulae.first
+    odie "No formula found!" unless formula
+
+    formula_path = formula.path
+    core = formula_path.to_s.include? "homebrew-core"
+    if ARGV.dry_run?
+      ohai "brew update"
+      ohai "brew tap homebrew/boneyard"
+      ohai "cp #{formula_path} #{HOMEBREW_LIBRARY}/Taps/homebrew/homebrew-boneyard"
+      ohai "git rm #{formula_path}"
+      ohai "Adding #{formula.name} to tab migrations" if core
+    else
+      safe_system "brew", "update"
+      safe_system "brew", "tap", "homebrew/boneyard"
+      cp formula.path, HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-boneyard"
+      formula.path.parent.cd { safe_system "git", "rm", formula.path }
+      if core
+        tap_migrations = Utils::JSON.load(File.read(HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-core/tap_migrations.json"))
+        tap_migrations[formula.name] = "homebrew/boneyard"
+        tap_migrations = tap_migrations.sort.inject({}) { |a, e| a.merge!(e[0] => e[1]) }
+        (HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-core/tap_migrations.json").atomic_write(JSON.pretty_generate(tap_migrations))
+      end
+    end
+    unless Formula["hub"].any_version_installed?
+      if ARGV.dry_run?
+        ohai "brew install hub"
+      else
+        safe_system "brew", "install", "hub"
+      end
+    end
```

Comment:
`hub` is a requirement anyway; it makes sense to me to install it at the very beginning before starting to manipulate various repositories. If installing `hub` fails for some reason, the repositories won't be left in an intermediate state.